lighthouse-core/audits/uses-rel-preconnect.js

```diff
@@ -113,17 +118,24 @@ class UsesRelPreconnectAudit extends Audit {
         return (record.startTime < firstRecord.startTime) ? record: firstRecord;
       });
 
-      const connectionTime =
-        firstRecordOfOrigin._timing.connectEnd - firstRecordOfOrigin._timing.dnsStart;
+      const securityOrigin = firstRecordOfOrigin.parsedURL.securityOrigin();
+
+      // Approximate the connection time with the duration of TCP (+potentially SSL) handshake
+      // DNS time can be large but can also be 0 if a commonly used origin that's cached, so make
+      // no assumption about DNS.
+      const additionalRtt = additionalRttByOrigin.get(securityOrigin) || 0;
+      let connectionTime = rtt + additionalRtt;
```

Comment:
is `rtt` not `number | undefined`? (based on `LH.Gatherer.Simulation.Options`, at least)

Reply:
oh, the file isn't type checked yet :) It would be good to handle it now, though

Reply:
fixed on the simulator side 👍